enterprise/models.py

```diff
@@ -2511,12 +2513,31 @@ class DefaultEnterpriseEnrollmentIntention(TimeStampedModel, SoftDeletableModel)
     )
     history = HistoricalRecords()
 
+    def get_cached_content_metadata(self):
```

Comment:
[consideration] Would it be helpful if this was a `@property`? E.g., `def content_metadata_for_content_key(self)`. I believe the benefit would be accessing the metadata via `self.content_metadata_for_content_key` vs. needing to call `self.get_cached_content_metadata()` each time.

The revised naming is just a suggestion :)